src/components/Card.js

```diff
@@ -1,37 +1,69 @@
 export default class Card {
-  constructor(data, cardSelector, handleImageClick) {
+  constructor(
+    data,
+    cardSelector,
+    handleImageClick,
+    handleLikeIcon,
+    handleDeleteCard
+  ) {
     this._name = data.name;
     this._link = data.link;
+    this.id = data._id;
+    this._isLiked = data.isLiked;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleLikeIcon = handleLikeIcon;
+    this.handleDeleteCard = handleDeleteCard;
   }
 
   _setEventListeners() {
     this._cardElement
-      .querySelector(".card__like-button")
+      .querySelector(".card__delete-button")
       .addEventListener("click", () => {
-        this._handleLikeButton();
+        this.handleDeleteCard(this);
       });
     this._cardElement
-      .querySelector(".card__delete-button")
+      .querySelector(".card__image")
       .addEventListener("click", () => {
-        this._handleCardDelete();
+        this._handleImagePreview(this._name, this._link);
       });
     this._cardElement
-      .querySelector(".card__image")
+      .querySelector(".card__like-button")
       .addEventListener("click", () => {
-        this._handleImagePreview();
+        this._handleLikeIcon(this);
       });
   }
 
-  _handleLikeButton() {
+  _handleLikeIcon() {
     this._cardElement
       .querySelector(".card__like-button")
```

Comment:
You've already found `this._likeButton` in line 78. Please, use it everywhere in `Card`

`Class fields` (the so-called **class variables** with `this`) are accessible inside any class method. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.